content/en/docs/k8s-operator/automatic.md

```diff
@@ -112,7 +112,67 @@ Coming Soon
 
 ### Java
 
-Coming Soon
+The following command creates a basic Instrumentation resource that is
+configured for instrumenting Java services.
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: demo-instrumentation
+spec:
+  exporter:
+    endpoint: http://demo-collector:4317
+  propagators:
+    - tracecontext
+    - baggage
+  sampler:
+    type: parentbased_traceidratio
+    argument: "1"
+EOF
+```
+
+By default, the Instrumentation resource that auto-instruments Java services
+uses `otlp` with the `grpc` protocol. This means that the configured endpoint
+must be able to receive OTLP over `grpc`. Therefore, the example uses
+`http://demo-collector:4317`, which connects to the `grpc` port of the
+otlpreceiver of the Collector created in the previous step.
+
+By default, the Java auto-instrumentation ships with
+[many instrumentation libraries](/docs/instrumentation/java/automatic/#supported-libraries-frameworks-application-services-and-jvms).
+This makes instrumentation easy, but could result in too much or unwanted data. If
+there are any libraries you do not want to use you can set the
+`OTEL_INSTRUMENTATION_[NAME]_ENABLED=false` where `[NAME]` is the name of the
+library. If you know exactly which libraries you want to use, you can disable the
+default libraries by setting `OTEL_INSTRUMENTATION_COMMON_DEFAULT_ENABLED=false`
+and then use `OTEL_INSTRUMENTATION_[NAME]_ENABLED=true` where `[NAME]` is the
+name of the library. For more details, see
+[Suppressing specific auto-instrumentation](/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-auto-instrumentation).
+
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: demo-instrumentation
+spec:
+  exporter:
+    endpoint: http://demo-collector:4317
+  propagators:
+    - tracecontext
+    - baggage
+  sampler:
+    type: parentbased_traceidratio
+    argument: "1"
```

Comment:
Can you link to the documentation for these configuration options? I'm not familiar with where this is defined or how it translates to the environment variable based config scheme the otel java agent used.

Reply:
For propagators the valid values are `tracecontext;baggage;b3;b3multi;jaeger;xray;ottrace;none` (https://github.com/open-telemetry/opentelemetry-operator/blob/1a12ac7bb23f78a39a1b6a7f216faa034342b2f7/apis/v1alpha1/propagators.go#L19).  The operator will add them to the `OTEL_PROPAGATORS` environment variable.

For sampler the valid values are `always_on, always_off, traceidratio, parentbased_always_on, parentbased_always_off, parentbased_traceidratio, jaeger_remote, xray` (always_on, always_off, traceidratio, parentbased_always_on, parentbased_always_off, parentbased_traceidratio, jaeger_remote, xray).  The operator will add the sampler and its args to `OTEL_TRACES_SAMPLER` and `OTEL_TRACES_SAMPLER_ARG`.

I agree that this page doesn't discuss those aspects clearly, and neither does the Operator docs.  I will get an operator docs PR created to address this so that I have something to link to from this doc.

Reply:
Is this blob required, or is this sorta a best practice?

Reply:
Are you asking if a propagator and sampler are required or only a best practice? They are best practices.

Reply:
@jack-berg since @TylerHelmuth is adding this content, are you satisfied with the changes for now? We can file a tracking issue to link to the config options (or pull them in here) separately.